g3/git/shell.py

```diff
@@ -0,0 +1,237 @@
+import asyncio
+import logging
+import os
+import shlex
+import subprocess
+import sys
+from typing import IO, Any, Dict, Optional, Sequence, Tuple, TypeVar, Union, overload
+
+# Shell commands generally return str, but with exitcode=True
+# they return a bool, and if stdout is piped straight to sys.stdout
+# they return None.
+_SHELL_RET = Union[bool, str, None]
+_HANDLE = Union[None, int, IO[Any]]
+
+
+def log_command(args: Sequence[str]) -> None:
+    """
+    Given a command, print it in a both machine and human readable way.
+
+    Args:
+        *args: the list of command line arguments you want to run
+        env: the dictionary of environment variable settings for the command
+    """
+    cmd = " ".join(shlex.quote(arg) for arg in args)
+    logging.info("$ " + cmd)
+
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+
+def merge_dicts(x: Dict[K, V], y: Dict[K, V]) -> Dict[K, V]:
+    z = x.copy()
+    z.update(y)
+    return z
+
+
+class Shell(object):
```

Comment:
```suggestion
class Shell:
```
object can be skipped in our version of python